packages/ckeditor-cloud-services-core/src/token/token.js

```diff
@@ -106,33 +112,72 @@ class Token {
 	 */
 	refreshToken() {
 		return this._refresh()
-			.then( value => this.set( 'value', value ) )
+			.then( value => {
+				this.set( 'value', value );
+				this._validateTokenValue();
```

Comment:
I would prefer to change this method to `this._validateTokenValue( value );` and call it before `this.set( 'value', value );`

We observe this property and send a request to the server when this value changed, so there is no sense to do it if the token is wrong - https://github.com/cksource/cs/blob/master/common/client/src/websocketgateway/websocketgateway.js#L431.

Reply:
Totally make sense. In addition, I've found that validating token in the init method makes no sense because the token, in that case, was already validated in the constructor.